python/src/equistore/operations/solve.py

```diff
@@ -75,7 +75,7 @@ def _solve_block(X: TensorBlock, Y: TensorBlock) -> TensorBlock:
     Y_values = Y.values.reshape(-1, Y_n_properties)
 
     if len(X.gradients_list()) > 0:
-        _check_same_gradients(X, Y, "solve")
+        _check_blocks(X, Y, ["gradients"], "solve")
```

Comment:
same, the if might not be required, and the loop below will do nothing if there are no gradients